csharp/ql/lib/semmle/code/csharp/frameworks/Format.qll

```diff
@@ -289,3 +289,31 @@ class FormatCall extends MethodCall {
     result = this.getArgument(this.getFirstArgument() + index)
   }
 }
+
+/**
+ * A method call to a method that parses a format string, for example a call
+ * to `string.Format()`.
+ */
+abstract private class FormatStringParseCallImpl extends MethodCall {
+  /**
+   * Gets the expression used as the format string.
+   */
+  abstract Expr getFormatExpr();
+}
+
+final class FormatStringParseCall = FormatStringParseCallImpl;
+
+private class OrdinaryFormatCall extends FormatStringParseCallImpl instanceof FormatCall {
```

Comment:
Perhaps just let `FormatCall` extend `FormatStringParseCallImpl` directly?

Reply:
Yes, that makes sense as well, I think. The last two commits are the changes since you reviewed last time.

Reply:
Ah, I misread your comment - and added the parse call as a FormatCall instead. Let me know, whether you think we should keep it like this (it does simplify the code).

Reply:
It looks like your rewrite changed the expected test output, is that good/expected?